Testscripts/Windows/VERIFY-XDP-MULTIPLE-NICS.ps1

```diff
@@ -0,0 +1,191 @@
+        -command $xdpCommand -RunInBackground -runAsSudo
+    return $testJob
+}
+
+function Main {
+    try {
+        $noClient = $true
+        $noServer = $true
+        foreach ($vmData in $allVMData) {
+            if ($vmData.RoleName -imatch "client") {
+                $clientVMData = $vmData
+                $noClient = $false
+            }
+            elseif ($vmData.RoleName -imatch "server") {
+                $noServer = $false
+                $serverVMData = $vmData
+            }
+        }
+        if ($noClient) {
+            Throw "No any client VM defined. Aborting Test."
+        }
+        if ($noServer) {
+            Throw "No any server VM defined. Aborting Test."
+        }
+
+        # CONFIGURE VM Details
+        Write-LogInfo "CLIENT VM details :"
+        Write-LogInfo "  RoleName : $($clientVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($clientVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($clientVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($clientVMData.InternalIP)"
+        Write-LogInfo "SERVER VM details :"
+        Write-LogInfo "  RoleName : $($serverVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($serverVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($serverVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($serverVMData.InternalIP)"
+
+        # Check for compatible kernel
+        $currentKernelVersion = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort `
+                -username $user -password $password -command "uname -r"
+        # ToDo: Update Minimum kernel version check once patches are in downstream distro.
+        if ((Compare-KernelVersion $currentKernelVersion $MIN_KERNEL_VERSION) -lt 0 -or $global:DetectedDistro -ne "UBUNTU"){
+            Write-LogInfo "Minimum kernel version required for XDP: $MIN_KERNEL_VERSION."`
+                "Unsupported kernel version: $currentKernelVersion or Unsupported distro: $($global:DetectedDistro)."
+            return $global:ResultSkipped
+        }
+
+        # PROVISION VMS
+        Provision-VMsForLisa -allVMData $allVMData -installPackagesOnRoleNames "none"
+
+        # Generate constants.sh and write all VM info into it
+        Write-LogInfo "Generating constants.sh ..."
+        $constantsFile = "$LogDir\constants.sh"
+        Set-Content -Value "# Generated by Azure Automation." -Path $constantsFile
+        Add-Content -Value "ip=$($clientVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "nicName=eth1" -Path $constantsFile
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+
+        # Build and Install XDP Dump application
+        $installXDPCommand = @"
+./XDPDumpSetup.sh 2>&1 > ~/xdpConsoleLogs.txt

```

Comment:
Can we use bash to execute the script. bash XDPDumpSetup.sh ? It would be good to source utils.sh first
For example ```bash ./XDPDumpSetup.sh > ~/xdpConsoleLogs.txt 2>&1```

Reply:
Doesn't the sh-bang at the start of the script [#!/bin/bash] takes care of that?

Reply:
Advantage of using bash is that you don't have to provide execute permission to the script.

Reply:
In this case, you can remove the line 103 and 104. Correct?

Reply:
fixed